File: MiniDOMSIP/Dialers/Bria/BriaDialer.cs
using System.IO;
using DMD.XML;

namespace minidom.PBX
{
    public class BriaDialer : DialerBaseClass
    {
        private string m_Path;

        public BriaDialer()
        {
            m_Path = GetBriaPath();
        }

        public override void Dial(string number)
        {
            if (!IsInstalled())
                return;
            DMD.IOUtils.Shell(m_Path + " -dial=sip:" + number);
        }

        public override bool IsInstalled()
        {
            return !string.IsNullOrEmpty(m_Path);
        }

        private static string GetBriaPath()
        {
            string p = Path.Combine(GetProgramFilesFolder(), @"CounterPath\Bria\bria.exe");
            if (File.Exists(p))
                return p;
            p = Path.Combine(GetProgramFilesFolder() + " (x86)", @"CounterPath\Bria\bria.exe");
            if (File.Exists(p))
                return p;
            return DMD.Strings.vbNullString;
        }

        public override string Name
        {
            get
            {
                return "CounterPath Bria";
            }
        }

        public override void HangUp()
        {
        }

        public override bool Equals(object obj)
        {
            if (!(obj is C3CXDialer))
                return false;
            return base.Equals(obj) && (((BriaDialer)obj).m_Path ?? "") == (m_Path ?? "");
        }
    }
}

File: MiniDOMSIP/Dialers/BriaDialer.vb
Imports System.IO

Public Class BriaDialer
    Inherits DialerBaseClass

    Private m_Path As String

    Public Sub New()
        Me.m_Path = GetBriaPath()
    End Sub

    Public Overrides Sub Dial(number As String)
        If Not Me.IsInstalled Then Return
        Shell(Me.m_Path & " -dial=sip:" & number)
    End Sub

    Public Overrides Function IsInstalled() As Boolean
        Return (Me.m_Path <> vbNullString)
    End Function

    Private Shared Function GetBriaPath() As String
        Dim p As String = Path.Combine(GetProgramFilesFolder, "CounterPath\Bria\bria.exe")
        If File.Exists(p) Then Return p
        p = Path.Combine(GetProgramFilesFolder() & " (x86)", "CounterPath\Bria\bria.exe")
        If File.Exists(p) Then Return p
        Return vbNullString
    End Function

    Public Overrides ReadOnly Property Name As String
        Get
            Return "CounterPath Bria"
        End Get
    End Property

    Public Overrides Sub HangUp()

    End Sub

    Public Overrides Function Equals(obj As Object) As Boolean
        If Not (TypeOf (obj) Is C3CXDialer) Then Return False
        Return MyBase.Equals(obj) AndAlso DirectCast(obj, BriaDialer).m_Path = Me.m_Path
    End Function

End Class
